### PR TITLE
SM-544 - DAT Group VIN decoding

### PR DESCRIPTION
**Pull request by rhvivancoeffio**
# Proposed Changes

### CODE REVIEW

File: internal/core/services/vin_decoding_service.go [hunk 1]
```diff
@@ -264,3 +289,30 @@ func buildFromDD(vin string, info *repoModel.DeviceDefinition) *models.VINDecodi
...
+	dossier := info.Body.GetDataVehicleIdentificationByVinResponse.VXS.Dossier[0].Vehicle
+	v := &models.VINDecodingInfoData{
+		VIN:  dossier.VINResult.VINVehicle.VINumber.VinCode,
+		Year: int32(dossier.BuildYear),
+		//Year:       2022,
+		Make:       strings.TrimSpace(dossier.ManufacturerName),
+		Model:      strings.TrimSpace(dossier.BaseModelName),
+		Source:     models.DATGroupProvider,
+		ExternalID: dossier.VINResult.VINVehicle.VINumber.VinCode,
+		//StyleName:  info.GetStyle(),
```
Comment (JamesReate) [inline internal/core/services/vin_decoding_service.go:308]: we'll need to think of something for the style name

File: internal/core/services/vin_decoding_service.go [hunk 2]
```diff
@@ -84,6 +88,15 @@ func (c vinDecodingService) GetVIN(ctx context.Context, vin string, dt *repoMode
 		if err != nil {
 			return nil, err
 		}
+	case models.DATGroupProvider:
```
Comment (JamesReate) [inline internal/core/services/vin_decoding_service.go:91]: looks good to me

**Review by JamesReate — COMMENTED** [review]
so far looks good, we'll just need to figure out what we use for the style, and tests

File: internal/core/services/vin_decoding_service.go (hunk 1 above)
Comment (rhvivancoeffio) [reply internal/core/services/vin_decoding_service.go:308]: What would be the fields to be able to map them?

**Pull request merged**